cms/static/js/views/utils/xblock_utils.js

```diff
@@ -310,6 +313,9 @@ function($, _, gettext, ViewUtils, ModuleUtils, XBlockInfo, StringUtils) {
         if (visibilityState === VisibilityState.staffOnly) {
             return 'is-staff-only';
         }
+        if (visibilityState === VisibilityState.hideFromTOC) {
```

Comment:
I know that this is not a refactor but I would change that function to something like this
```
 let visibleStates = { 
  [VisibilityState.staffOnly]: 'is-staff-only',
  [VisibilityState.hideFromTOC]: 'is-hidden-from-toc',
  [VisibilityState.gated]: 'is-gated',
  [VisibilityState.live]: 'is-live',
  [VisibilityState.ready]: 'is-ready',
  [VisibilityState.needsAttention]: 'has-warnings'
}

return visibleStates[visibilityState] || '';

```

Reply:
Thanks for the suggestion! Although I'd like to implement this refactoring for better readability, I will not do it in this PR. Please open a PR with the suggestion if you have time. We'd appreciate it!